### ISSUE TITLE
[bug]: why is user tags not added or linked on subscriptions

### ISSUE THREAD

**Issue by aman123shi** [OneSignal/react-native-onesignal#1618]
### How can we help?

I am creating users with tags using external_id as an identifier by using the REST API from my backend and the request is successful but when a player subscribes for push notification from React native app the user tags are not added to the user subscriptions but after the user subscribed and I try to add that user again the user tags are added to the subscriptions how to fix this

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by jennantilla** [comment]
Hi @aman123shi thanks for reaching out! Just to clarify, you are saying that when using the SDK to add tags, the tags are not added until the user subscribes? Would you be able to send a [log](https://documentation.onesignal.com/docs/capturing-a-debug-log) from the device capturing this behavior, please?

**Comment by aman123shi** [comment]
no, am not setting tags from the SDK  I am saying I created a user with tags and external_id using Users rest API from my backend when a user registers to my system so when a user from the mobile app tries to subscribe to push notification using that same exteranl_id that I set when my user registers it's tags are not added to his subscriptions when I see it from the oneSignal dashboard . by the way I am using react-native-onsignal 4.5.2 since the version 5 does not support identity verification

**Comment by JovanMwesigwa** [comment]
Anyone figured how to solve this?

**Comment by fadi-george** [comment]
Hi all, apologies this issue has gone stale; the team is no longer actively supporting and patching the v4 major version of this SDK.
I recommend to upgrade to v5 of our SDK. Please see our [documentation](https://documentation.onesignal.com/docs/en/user-model-migration-guide) on the new user model and migration steps, and let us know if we can help you with the transition.
